feature/bgp/otg_tests/bgp_override_as_path_split_horizon_test/bgp_override_as_path_split_horizon_test.go

```diff
@@ -0,0 +1,581 @@
+	}
+	if gotSent := gnmi.Get(t, dut, prefixesv4.Sent().State()); gotSent != wantSent {
+		t.Errorf("Sent prefixes mismatch: got %v, want %v", gotSent, wantSent)
+	}
+}
+
+// configreRoutePolicy adds route-policy config.
+func configureRoutePolicy(t *testing.T, dut *ondatra.DUTDevice, name string, pr oc.E_RoutingPolicy_PolicyResultType) {
+	d := &oc.Root{}
+	rp := d.GetOrCreateRoutingPolicy()
+	pd := rp.GetOrCreatePolicyDefinition(name)
+	st, err := pd.AppendNewStatement("id-1")
+	if err != nil {
+		t.Fatal(err)
+	}
+	stc := st.GetOrCreateConditions()
+	stc.InstallProtocolEq = oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP
+	st.GetOrCreateActions().PolicyResult = pr
+	gnmi.Replace(t, dut, gnmi.OC().RoutingPolicy().Config(), rp)
+}
+
+// verifyOTGPrefixTelemetry is to Validate prefix received on OTG por2.
+func verifyOTGPrefixTelemetry(t *testing.T, otg *otg.OTG, wantPrefix bool) {
+	t.Helper()
+	_, ok := gnmi.WatchAll(t, otg, gnmi.OTG().BgpPeer(atePort2.Name+".BGP4.peer").UnicastIpv4PrefixAny().State(),
+		time.Minute, func(v *ygnmi.Value[*otgtelemetry.BgpPeer_UnicastIpv4Prefix]) bool {
+			return v.IsPresent()
+		}).Await(t)
+
+	if ok {
+		bgpPrefixes := gnmi.GetAll(t, otg, gnmi.OTG().BgpPeer(atePort2.Name+".BGP4.peer").UnicastIpv4PrefixAny().State())
+		for _, prefix := range bgpPrefixes {
+			if prefix.GetAddress() == advertisedRoutesv4Net {
+				if wantPrefix {
+					gotASPath := prefix.AsPath[len(prefix.AsPath)-1].GetAsNumbers()
+					t.Logf("Received prefix %v on otg as expected with AS-PATH %v", prefix.GetAddress(), gotASPath)
+				} else {
+					t.Errorf("Prefix %v is not received on otg", prefix.GetAddress())
+				}
+			}
+		}
+	}
+}
+
+func testSplitHorizonNoAllowOwnIn(t *testing.T, args *otgTestArgs) {
+	t.Log("Baseline Test No allow-own-in")
+
+	t.Log("Advertise a prefix from the ATE with an AS-path that includes AS dutLocalAS1 (DUT's AS) in the middle (e.g., AS-path: 65500 dutLocalAS1 65499")
+	advBGPRouteFromOTG(t, args, []uint32{65500, dutLocalAS1, 65499})
+
+	t.Log("Validate session state and capabilities received on DUT using telemetry.")
+	verifyBgpTelemetry(t, args.dut, []*bgpNeighbor{nbr1, nbr2})
+	verifyBGPCapabilities(t, args.dut, []*bgpNeighbor{nbr1, nbr2})
+
+	t.Log("Verify that the ATE Port2 doesn't receive the route. due to the presence of its own AS in the path.")
+	verifyPrefixesTelemetry(t, args.dut, nbr1.neighborip, 0, 0)
+	verifyPrefixesTelemetry(t, args.dut, nbr2.neighborip, 0, 0)
+	verifyOTGPrefixTelemetry(t, args.otg, false)
+}
+
+func testSplitHorizonAllowOwnAs1(t *testing.T, args *otgTestArgs) {
+	t.Log("Test allow-own-as 1, Enable allow-own-as 1 on the DUT.")
+	dutConfPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(args.dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP")
+	if deviations.BgpAllowownasDiffDefaultValue(args.dut) {
+		gnmi.Replace(t, args.dut, dutConfPath.Bgp().PeerGroup(peerGrpName1).AsPathOptions().AllowOwnAs().Config(), 2)
```

Comment:
I'm not inclined to allow this deviation?  Why doesn't "1" work as specified in the OC Model?  Is juniper going to be fixing this soon?

```
/network-instances/network-instance/protocols/protocol/bgp/neighbors/neighbor/as-path-options/config/allow-own-as

description:
Specify the number of occurrences of the local BGP speaker's AS that can occur within the AS_PATH before it is rejected.
nodetype: leaf (rw)
type: uint8
default: 0
```

Reply:
We have different default values and it is updated in latest releases to match with OC default value.